Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/auditrepository/hibernate/AuditRepositoryDAO.java

```diff
@@ -161,4 +167,189 @@ public List queryAuditRepositoryOnCriteria(String eUserId, String ePatientId, Da
+            queryList = queryCriteria.list();
+
+        } catch (HibernateException e) {
+            LOG.error("Exception in AuditLog.get() occurred due to :" + e.getLocalizedMessage(), e);
+        } finally {
+            // Actual contact insertion will happen at this step
+            if (session != null) {
+                session.flush();
+                session.close();
+            }
+        }
+        return queryList;
+    }
+
+    /**
+     * This method does a query to database to get the Audit Log Messages based different options
+     *
+     * @param outcome
+     * @param startDate
+     * @param userId
+     * @param eventTypeList
+     * @param remoteHcidList
+     * @param endDate
+     * @return List
+     */
+    public List queryByAuditValues(Integer outcome, List<String> eventTypeList, String userId, List<String> remoteHcidList, String startDate, String endDate) {
+
+        Session session = null;
+        List<AuditRepositoryRecord> queryList = null;
+        try {
+            SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+            session = sessionFactory.openSession();
+
+            LOG.info("Getting Record for Audit Viewer ");
+
+            // Build the criteria
+            Criteria queryCriteria = session.createCriteria(AuditRepositoryRecord.class);
+
+            if (outcome != null && outcome >= 0) {
+                queryCriteria.add(Restrictions.eq("outcome", outcome));
+            }
+
+            if (NullChecker.isNotNullish(eventTypeList)) {
+                queryCriteria.add(Restrictions.in("eventType", eventTypeList));
+            }
+
+            if (NullChecker.isNotNullish(userId)) {
+                queryCriteria.add(Restrictions.eq("userId", userId));
+            }
+
+            if (NullChecker.isNotNullish(remoteHcidList)) {
+                queryCriteria.add(Restrictions.in("remoteHcid", remoteHcidList));
+            }
+
+            if (NullChecker.isNotNullish(startDate) && NullChecker.isNotNullish(endDate)) {
+                SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy");
+
+                queryCriteria.add(Restrictions.between("eventTimestamp", new Date(sdf.parse(startDate).getTime()),
+                    new Date(sdf.parse(endDate).getTime())));
+
+            } else if (NullChecker.isNotNullish(startDate) && NullChecker.isNullish(endDate)) {
+                SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy");
+
+                queryCriteria.add(Restrictions.ge("eventTimestamp", new Date(sdf.parse(startDate).getTime())));
+
+            } else if (NullChecker.isNullish(startDate) && NullChecker.isNotNullish(endDate)) {
+                SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy");
+
+                queryCriteria.add(Restrictions.le("eventTimestamp", new Date(sdf.parse(endDate).getTime())));
+
+            }
+
+            queryCriteria.setProjection(Projections.projectionList()
```

Comment:
This set project is repeated in both methods and could be pulled out into a separate method.